test/routes/src/server/routes/t_launchpad.js

```diff
@@ -569,6 +570,214 @@ describe('The Launchpad API endpoint', () => {
+                'build_log_url': `${lp_api_url}${lp_snap_path}/+build/12345/+files/buildlog_snap_ubuntu_xenial_armhf_${lp_snap_name}_BUILDING.txt.gz`,
+                'arch_tag': 'armhf',
+              }
+            ]
+          });
+      });
+
+      afterEach(() => {
+        nock.cleanAll();
+      });
+
+      it('should return a 200 OK response', (done) => {
+        supertest(app)
+          .get('/launchpad/builds')
+          .query({ snap_link: lp_snap_link })
+          .expect(200, done);
+      });
+
+      it('should return a "success" status', (done) => {
+        supertest(app)
+          .get('/launchpad/builds')
+          .query({ snap_link: lp_snap_link })
+          .expect(hasStatus('success'))
+          .end(done);
+      });
+
+      it('should return body with "snap-builds-found" message', (done) => {
+        supertest(app)
+          .get('/launchpad/builds')
+          .query({ snap_link: lp_snap_link })
+          .expect(hasMessage('snap-builds-found'))
+          .end(done);
+      });
+
+      it('should return builds list as payload message', (done) => {
+        supertest(app)
+          .get('/launchpad/builds')
+          .query({ snap_link: lp_snap_link })
+          .end((err, res) => {
+            expect(res.body.payload.message.length).toEqual(1);
+            done(err);
+          });
+      });
+
+    });
+
+    context('when builds fail to fetch', () => {
+
+      beforeEach(() => {
+        // when getting snap data from API (via self_link)
+        nock(lp_api_url)
+          .get(lp_snap_path)
+          .reply(200, {
+            name: lp_snap_name,
+            builds_collection_link: `${lp_api_url}${lp_builds_path}`
+          });
+
+        // when getting builds list (via builds_collection_link)
+        nock(lp_api_url)
+          .get(lp_builds_path)
+          .query({
+            'ws.start': 0,
+            'ws.size': 10
+          })
+          .reply(404, 'Not found');
+      });
+
+      afterEach(() => {
+        nock.cleanAll();
+      });
+
+      it('should return a 404 response', (done) => {
+        supertest(app)
+          .get('/launchpad/builds')
+          .query({ snap_link: lp_snap_link })
+          .expect(404, done);
+      });
+
+      it('should return a "error" status', (done) => {
+        supertest(app)
+          .get('/launchpad/builds')
+          .query({ snap_link: lp_snap_link })
+          .expect(hasStatus('error'))
+          .end(done);
+      });
+
+      it('should return body with error message', (done) => {
+        supertest(app)
+          .get('/launchpad/builds')
+          .query({ snap_link: lp_snap_link })
+          .expect(hasMessage('lp-error', 'Not found'))
+          .end(done);
+      });
+
+    });
+
+    context('when snap fails to fetch', () => {
+
+      beforeEach(() => {
+        // when getting snap data from API (via self_link)
+        nock(lp_api_url)
+          .get(lp_snap_path)
+          .reply(404, 'Not found');
```

Comment:
@cjwatson  Are there any specific scenarios/cases we would like to cover here?

What possible errors can we get when trying to get snap?